apps/core/utils.py

```diff
@@ -0,0 +1,36 @@
+import random
+import string
+
+
+ALPHANUMERIC_CHARS = string.ascii_lowercase + string.digits
+STRING_LENGTH = 6
+
+# function that generates a random string
+
+def generate_random_string(
+                            chars=ALPHANUMERIC_CHARS, 
+                            length=STRING_LENGTH
+                            ):
+    return "".join(random.choice(chars) for _ in range(length))
+
+
+# a function that generates unique slug for every 
+# item posted on freebay
+
+def generate_unique_slug(word):
+
+    alphanumeric = ''
+
+    for character in word:
+        
+        if character == " ":
+            alphanumeric += "-"
+
+        if character == "-":
+            pass
+
+        if character.isalnum():
+            alphanumeric += character
+
+    return alphanumeric + generate_random_string()
```

Comment:
Remove this function and just use Python's [`replace()`](https://www.w3schools.com/python/ref_string_replace.asp):

```python
some_str = "hello there"
some_str.replace(" ", "-") # replaces spaces with dashes
```

Reply:
@gmelodie thanks the .replace would actually make my work easier. I would need your input on this though:
Yeah, so I'm generating the slug from an item's title, and even though it's very unlikely, I imagine that some titles might have special characters. For example: "You need a Spanish dictionary?". So when generating slugs I'm not just removing white spaces, I'm also trying to remove special characters.
Any idea of how I could achieve this with the .replace?